### PR TITLE
add distributed pi code

### PR DESCRIPTION
**Pull request by bleything**
this branch contains a nodejs implementation of the [Bailey-Borwein-Plouffe formula](https://en.wikipedia.org/wiki/Bailey%E2%80%93Borwein%E2%80%93Plouffe_formula). It's based heavily on the [python version at literateprograms.org](https://literateprograms.org/pi_with_the_bbp_formula__python_.html).

It also contains very minimal HTTP and PubSub entrypoints for [Cloud Functions](https://cloud.google.com/functions) using the [Functions Framework](https://github.com/GoogleCloudPlatform/functions-framework-nodejs).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -70,6 +70,7 @@ vendor/
 
 
 # Node
+node_modules/
```
Comment (yuryu) [inline .gitignore:73]: It's already in line 115.

File: distpi/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+Distributed Pi
+================================================================================
+
+This directory contains an implementation of the [Bailey-Borwein-Plouffe formula]
```
Comment (yuryu) [inline distpi/README.md:4]: [Bailey-Borwein-Plouffe formula] or [[Bailey-Borwein-Plouffe]
not linked

File: distpi/index.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    let result = calculatePiChunk(offset, length);
+    res.send(result);
+});
+
+exports.pubsubCalc = (evt, ctx) => {
+    const message = Buffer.from(evt.data, "base64").toString();
+    const params  = JSON.parse(message);
+
+    let offset = params.start || 0;
+    let length = params.size  || 100;
```
Comment (yuryu) [inline distpi/index.js:31]: ditto

File: distpi/index.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const functions = require('@google-cloud/functions-framework');
+const calculatePiChunk = require("./bbp");
+
+functions.http('httpCalc', (req, res) => {
+    let offset = req.query.offset || req.body.offset || 0;
+    let length = req.query.length || req.body.length || 100;
```
Comment (yuryu) [inline distpi/index.js:20]: use const here

**Review by yuryu — APPROVED** [review]
lgtm with nits

File: .gitignore (hunk 1 above)
Comment (bleything) [reply .gitignore]: rebased out this change

File: distpi/README.md (hunk 2 above)
Comment (bleything) [reply distpi/README.md]: doh. fixed.

File: distpi/index.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+    let result = calculatePiChunk(offset, length);
```
Comment (bleything) [inline distpi/index.js:22]: @yuryu should this be const too?

Comment (yuryu) [reply distpi/index.js]: yup, as long as they're not modified after initialization. I'm not sure why return is not returned but I'm a Functions newbie.

Comment (bleything) [reply distpi/index.js]: 👍🏻

**Pull request merged**